lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -1319,6 +1319,18 @@ static bool isOkToBitSelectFrom(Value v) {
   return true;
 }
 
+/// Return true for operations that are always inlined.
+static bool isExpressionAlwaysInline(Operation *op) {
+  if (isa<ConstantOp>(op) || isa<ArrayIndexInOutOp>(op))
+    return true;
```

Comment:
This makes sense to me. This means that any usage of an `ArrayIndexInOutOp` will always be linlined, regardless of LHS or RHS usage, yes?

Originally, I had thought that this should be differentiated and only LHS usage should be inlined. However, while inlining LHS usages are necessary for correctness, inlining RHS usages is a good idea for readability.

Reply:
yea, this is how it's been working, now I just don't exclude array indexes in a different blocks.

Reply:
cool thanks for fixing this.  If this logic gets too insane, we can consider duplicating these things out in the prepass over the module, similar to how variadic ops are split into binary ops.